src/gen_hook.erl

```diff
@@ -70,13 +71,17 @@
                        hook_fn :: hook_fn(),
                        extra :: extra()}).
 
--define(TABLE, ?MODULE).
 %%%----------------------------------------------------------------------
 %%% API
 %%%----------------------------------------------------------------------
 start_link() ->
     gen_server:start_link({local, ?MODULE}, ?MODULE, [], []).
 
+start_link_with_reload() ->
```

Comment:
Just a note: this function could be placed in a small test helper, because both functions are exported.